merging a couple PRs

c0sc0 opened the pull request:
merges #32 and #33 

- flask_freezer to generate static page
- flask_server to run it yourself (dynamic stuff coming later)
- trying out the 6 month window
- [webpage redesign](https://i.imgur.com/iJEBfpN.png) & sortable/searchable table

NihilistPenguin committed:
Update market_scanner.py

Change time output to minutes instead of seconds
Add ability to print all results together an the end
Output anomalous tickers to a results.txt file

NihilistPenguin committed:
Add files via upload

opens anomalous tickers in Robinhood in a new browser window

NihilistPenguin committed:
Update grapher.py

Creates graphs for all tickers in results.txt 
Allows option for single ticker or tickers of your choice

NihilistPenguin committed:
Add files via upload

Example results.txt file

NihilistPenguin committed:
Update README.md

change line numbers for controlling the script

NihilistPenguin committed:
Update README.md

Add/update descriptions grapher.py and openRobinhood.py

NihilistPenguin committed:
Update README.md

Fix formatting error

NihilistPenguin committed:
Update grapher.py

Provides option to save graphs to the figures directory or to print them to screen

NihilistPenguin committed:
Update grapher.py

Create figures dir if it doesn't exist

NihilistPenguin committed:
Update market_scanner.py

remove previous addition of printing anomalous results together at the end

Commit:
add figures directory with demo images

Commit:
add figures directory with sample images

NihilistPenguin committed:
Delete .gitignore

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Merge pull request #16 from tcosculluela/master

Parallelization

SamPom100 committed:
Revert "Parallelization"

Commit:
Merge pull request #20 from SamPom100/revert-16-master

Revert "Parallelization"

Commit:
Merge pull request #18 from NihilistPenguin/master

Add various functionality

SamPom100 committed:
Revert "Add various functionality "

Commit:
Merge pull request #21 from SamPom100/revert-18-master

Revert "Add various functionality "

Commit:
Make Anomaly Finding more Pandas friendly

This removes the unused finding function, and replaces it with the
normal algorithm. Then, all the indexing through the Pandas Dataframes
are refactored in preference for the faster Pandas calls.

Commit:
Merge branch 'master' into pandas_branch

Commit:
Merge pull request #1 from alexshirley/pandas_branch

Pandas branch

Commit:
Rebasing off of updated upstream

Commit:
Update README.md

Remove additional /r in README

c0sc0 committed:
merge success

flask_freezer to generate the static page

flask_server to host it yourself (dynamic stuff coming later)

6 month window

elizabethshirley commented on market_scanner.py:
```diff
@@ -23,63 +29,43 @@ def getData(self, ticker):
-        data = yf.download(ticker, pastDate, currentDate)
+
+        #data = yf.download(ticker, pastDate, currentDate)
+        data = yf.download(tickers = ticker,period='6mo')
```
Instead of commenting out, let's straight up remove the line 33 and 35, as well as the `currentDate` and `pastDate` variables. Then, let's pass in a parameter, and provide a command line argument for it.

Gotta keep it clean

c0sc0 replied:
hahaha yeah, it was 3AM and I couldn't be arsed to do it right. Luckily you came in clutch

elizabethshirley reviewed (changes requested):
Let's be mindful to keep the code-base as clean as possible. Most people coming here likely don't know what to do with comments, when a command line argument would suffice.

Also, don't PR *.pyc files
https://coderwall.com/p/wrxwog/why-not-to-commit-pyc-files-into-git-and-how-to-fix-if-you-already-did

Commit:
Merge branch 'master' into Pandas-Refactor-updated-to-master

SamPom100 commented:
thanks for all of this. I didn't see your PRs until I was done using freezer to make a static web page.

here it is: https://sampom100.github.io/UnusualVolumeDetector/

Commit:
Merge branch 'master' into Pandas-Refactor-updated-to-master

Commit:
Catching up with changes on master

Commit:
Allow modifiable DAY_CUTOFF

c0sc0 committed:
Merge pull request #2 from alexshirley/Pandas-Refactor-updated-to-master

merge & clean

elizabethshirley commented on market_scanner.py:
```diff
@@ -8,26 +8,30 @@
+###########################
+
+# Change variables to your liking then run the script
+MONTH_CUTTOFF = 6
```
Instead of running around with Globals, we probably want to have a class called config or something. On startup, it's populated with the default values seen here, unless there are arguments

https://www.tutorialspoint.com/python/python_command_line_arguments.htm

This allows you to have one script running with the defaults and the other running with some other behavior entirely, without needing two copies of the repo.

elizabethshirley commented on stocklist.py:
```diff
@@ -2,6 +2,10 @@
+# this is used to get all tickers from the market.
+
+
 exportList = []
```
There was a pull request which put the export list in a class and made it a return of the script. I think that's a very good idea, because like my other comment, allows for greater flexibility.

elizabethshirley commented on website_generator.py:
```diff
@@ -0,0 +1,43 @@
+
+app = flask.Flask(__name__, static_url_path='')
+app.config["DEBUG"] = False
+app.config['SECRET_KEY'] = 'deditaded wam'
```
There's got to be some way to not have your secrets in plain text on a public repository

c0sc0 replied:
haha yeah, I pretty much completely ripped the flask script from another project and noticed this

maybe i should add more deditated wam to the server

c0sc0 committed:
merge & clean

template.html for generating static page with website_generator

dynamic.html for serving yourself with flask_server

refactored globals to class variables in market_scanner

print all results at end when running market_scanner

line 34 fixes the off-by-1 bug (if it exists, will test it on Monday).

elizabethshirley reviewed (changes requested):
In addition to my other comments, you likely want to add Git LFS for all the graphics files, and move them there. Whenever you slightly tweak a graphic, the entire graphic is stored on Git, which slows down pulling as the project wears on.

c0sc0 committed:
hide secret key, default params, exportList in stocklist

try to pull secret key from env var

added default parameters,  now these are set when you instantiate the object

made exportList a class variable

elizabethshirley reviewed (approved):
geil!

c0sc0 commented:
continued in #40
